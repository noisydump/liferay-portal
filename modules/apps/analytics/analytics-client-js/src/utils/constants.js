/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

export const DEBOUNCE = 1500;

export const FLUSH_INTERVAL = 2000;

// Limit of a queue localStorage size in kilobytes.

export const QUEUE_STORAGE_LIMIT = 512;

// Local Storage keys

export const STORAGE_KEY_CONTEXTS = 'ac_client_context';

export const STORAGE_KEY_EVENTS = 'ac_client_batch';

export const STORAGE_KEY_IDENTITY = 'ac_client_identity';

export const STORAGE_KEY_MESSAGES = 'ac_message_queue';

export const STORAGE_KEY_USER_ID = 'ac_client_user_id';

// Request Constants

export const LIMIT_FAILED_ATTEMPTS = 7;

export const REQUEST_TIMEOUT = 5000;
