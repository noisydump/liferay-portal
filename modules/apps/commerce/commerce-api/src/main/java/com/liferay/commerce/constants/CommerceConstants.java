/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.constants;

/**
 * @author Andrea Di Giorgi
 * @author Alessio Antonio Rendina
 * @author Marco Leo
 */
public class CommerceConstants {

	/**
	 * @deprecated As of Athanasius (7.3.x)
	 */
	@Deprecated
	public static final String ORDER_FIELDS_SERVICE_NAME =
		"com.liferay.commerce.order.fields";

	/**
	 * @deprecated As of Athanasius (7.3.x)
	 */
	@Deprecated
	public static final String ORDER_SERVICE_NAME =
		"com.liferay.commerce.order";

	/**
	 * @deprecated As of Athanasius (7.3.x)
	 */
	@Deprecated
	public static final String PRICE_SERVICE_NAME =
		"com.liferay.commerce.price";

	public static final String RESOURCE_NAME_ADDRESS =
		"com.liferay.commerce.address";

	public static final String RESOURCE_NAME_AVAILABILITY =
		"com.liferay.commerce.availability";

	public static final String RESOURCE_NAME_HEALTH =
		"com.liferay.commerce.health";

	public static final String RESOURCE_NAME_SHIPMENT =
		"com.liferay.commerce.shipment";

	public static final String RESOURCE_NAME_SUBSCRIPTION =
		"com.liferay.commerce.subscription";

	public static final String SERVICE_NAME_ORDER =
		"com.liferay.commerce.order";

	public static final String SERVICE_NAME_ORDER_FIELDS =
		"com.liferay.commerce.order.fields";

	public static final String SERVICE_NAME_PRICE =
		"com.liferay.commerce.price";

	public static final String SERVICE_NAME_SHIPPING =
		"com.liferay.commerce.shipping";

	public static final String SERVICE_NAME_TAX = "com.liferay.commerce.tax";

	/**
	 * @deprecated As of Athanasius (7.3.x)
	 */
	@Deprecated
	public static final String SHIPPING_SERVICE_NAME =
		"com.liferay.commerce.shipping";

	/**
	 * @deprecated As of Athanasius (7.3.x)
	 */
	@Deprecated
	public static final String TAX_SERVICE_NAME = "com.liferay.commerce.tax";

}