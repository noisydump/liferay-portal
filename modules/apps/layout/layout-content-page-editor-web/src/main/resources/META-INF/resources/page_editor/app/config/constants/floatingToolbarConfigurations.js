/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {CollectionConfigurationPanel} from '../../../app/components/floating-toolbar/CollectionConfigurationPanel';
import {ContainerConfigurationPanel} from '../../../app/components/floating-toolbar/ContainerConfigurationPanel';
import {FragmentConfigurationPanel} from '../../../app/components/floating-toolbar/FragmentConfigurationPanel';
import {ImagePropertiesPanel} from '../../../app/components/floating-toolbar/ImagePropertiesPanel';
import LinkPanel from '../../../app/components/floating-toolbar/LinkPanel';
import {MappingPanel} from '../../../app/components/floating-toolbar/MappingPanel';
import {RowConfigurationPanel} from '../../../app/components/floating-toolbar/RowConfigurationPanel';
import {EDITABLE_FLOATING_TOOLBAR_BUTTONS} from './editableFloatingToolbarButtons';
import {LAYOUT_DATA_FLOATING_TOOLBAR_BUTTONS} from './layoutDataFloatingToolbarButtons';

const {
	collectionConfiguration,
	containerConfiguration,
	fragmentConfiguration,
	rowConfiguration,
} = LAYOUT_DATA_FLOATING_TOOLBAR_BUTTONS;

const {imageProperties, link, map} = EDITABLE_FLOATING_TOOLBAR_BUTTONS;

export const FLOATING_TOOLBAR_CONFIGURATIONS = {
	[collectionConfiguration.panelId]: CollectionConfigurationPanel,
	[containerConfiguration.panelId]: ContainerConfigurationPanel,
	[fragmentConfiguration.panelId]: FragmentConfigurationPanel,
	[imageProperties.panelId]: ImagePropertiesPanel,
	[link.panelId]: LinkPanel,
	[map.panelId]: MappingPanel,
	[rowConfiguration.panelId]: RowConfigurationPanel,
};
