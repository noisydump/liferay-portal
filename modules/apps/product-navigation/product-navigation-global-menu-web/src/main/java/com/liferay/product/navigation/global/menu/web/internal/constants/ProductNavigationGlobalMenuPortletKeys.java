/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.product.navigation.global.menu.web.internal.constants;

/**
 * @author Eudaldo Alonso
 */
public class ProductNavigationGlobalMenuPortletKeys {

	public static final String PRODUCT_NAVIGATION_GLOBAL_MENU =
		"com_liferay_product_navigation_global_apps_web_internal_portlet_" +
			"ProductNavigationGlobalMenuPortlet";

}