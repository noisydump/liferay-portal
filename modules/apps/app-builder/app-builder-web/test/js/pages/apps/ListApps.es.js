/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {waitForElementToBeRemoved} from '@testing-library/dom';
import {cleanup, render} from '@testing-library/react';
import React from 'react';

import ListApps from '../../../../src/main/resources/META-INF/resources/js/pages/apps/ListApps.es';
import * as time from '../../../../src/main/resources/META-INF/resources/js/utils/time.es';
import AppContextProviderWrapper from '../../AppContextProviderWrapper.es';
import {RESPONSES} from '../../constants.es';

describe('ListApps', () => {
	beforeEach(() => {
		jest.spyOn(time, 'fromNow').mockImplementation(() => 'months ago');
	});

	afterEach(() => {
		cleanup();
		jest.restoreAllMocks();
	});

	it('renders', async () => {
		fetch.mockResponse(JSON.stringify(RESPONSES.ONE_ITEM));

		const {asFragment} = render(
			<ListApps match={{params: {dataDefinitionId: ''}, url: '/'}} />,
			{wrapper: AppContextProviderWrapper}
		);

		await waitForElementToBeRemoved(() =>
			document.querySelector('span.loading-animation')
		);

		expect(asFragment()).toMatchSnapshot();
	});

	it('renders with dataDefinitionId and 5 apps in the list', async () => {
		fetch.mockResponse(JSON.stringify(RESPONSES.MANY_ITEMS(5)));

		const {container} = render(
			<ListApps match={{params: {dataDefinitionId: '1'}, url: '/'}} />,
			{wrapper: AppContextProviderWrapper}
		);

		await waitForElementToBeRemoved(() =>
			document.querySelector('span.loading-animation')
		);

		expect(container.querySelector('tbody').children.length).toEqual(5);
	});

	it('renders with no dataDefinitionId and 5 apps in the list', async () => {
		fetch.mockResponse(JSON.stringify(RESPONSES.MANY_ITEMS(5)));

		const {container} = render(
			<ListApps match={{params: {dataDefinitionId: ''}, url: '/'}} />,
			{wrapper: AppContextProviderWrapper}
		);

		await waitForElementToBeRemoved(() =>
			document.querySelector('span.loading-animation')
		);

		expect(container.querySelector('tbody').children.length).toEqual(5);
	});

	it('renders with empty state', async () => {
		fetch.mockResponse(JSON.stringify(RESPONSES.NO_ITEMS));

		const {container} = render(
			<ListApps match={{params: {dataDefinitionId: ''}, url: '/'}} />,
			{wrapper: AppContextProviderWrapper}
		);

		await waitForElementToBeRemoved(() =>
			document.querySelector('span.loading-animation')
		);

		expect(
			container.querySelector('.taglib-empty-result-message-title')
				.textContent
		).toEqual('there-are-no-apps-yet');
	});
});
