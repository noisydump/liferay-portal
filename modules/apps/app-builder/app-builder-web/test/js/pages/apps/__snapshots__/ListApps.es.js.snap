// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListApps renders 1`] = `
<DocumentFragment>
  <div
    class="tools-control-group"
  >
    <div
      class="control-menu-level-1-heading"
    />
  </div>
  <nav
    class="management-bar navbar navbar-expand-md management-bar-light"
  >
    <div
      class="container-fluid container-fluid-max-xl"
    >
      <div
        class="navbar-form navbar-form-autofit navbar-overlay navbar-overlay-sm-down"
      >
        <div
          class="container-fluid container-fluid-max-xl"
        >
          <form
            role="search"
          >
            <div
              class="input-group"
            >
              <div
                class="input-group-item"
              >
                <input
                  aria-label="search"
                  class="form-control input-group-inset input-group-inset-after"
                  placeholder="search..."
                  type="text"
                  value=""
                />
                <span
                  class="input-group-inset-item input-group-inset-item-after"
                >
                  <button
                    class="navbar-breakpoint-d-none btn btn-monospaced btn-unstyled"
                    type="button"
                  >
                    <svg
                      class="lexicon-icon lexicon-icon-times"
                      role="presentation"
                    >
                      <use
                        xlink:href="#times"
                      />
                    </svg>
                  </button>
                  <button
                    class="btn btn-monospaced btn-unstyled"
                    type="submit"
                  >
                    <svg
                      class="lexicon-icon lexicon-icon-search"
                      role="presentation"
                    >
                      <use
                        xlink:href="#search"
                      />
                    </svg>
                  </button>
                </span>
              </div>
            </div>
          </form>
        </div>
      </div>
      <ul
        class="navbar-nav"
      >
        <li
          class="nav-item navbar-breakpoint-d-none"
        >
          <button
            class="nav-link nav-link-monospaced btn btn-unstyled"
            type="button"
          >
            <svg
              class="lexicon-icon lexicon-icon-search"
              role="presentation"
            >
              <use
                xlink:href="#search"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
  </nav>
  <div
    class="container-fluid container-fluid-max-xl"
  >
    <div>
      <div
        class="table-responsive"
      >
        <table
          class="table table-autofit table-list"
        >
          <thead>
            <tr
              class=""
            >
              <th
                class="table-cell-expand text-left"
              >
                name
              </th>
              <th
                class="table-cell-expand-smaller text-left"
              >
                object
              </th>
              <th
                class="table-cell-expand-smaller text-left"
              >
                deployed-as
              </th>
              <th
                class="table-cell-expand-smaller text-left"
              >
                create-date
              </th>
              <th
                class="table-cell-expand-smaller text-left"
              >
                modified-date
              </th>
              <th
                class="table-cell-expand-smaller text-left"
              >
                status
              </th>
              <th
                class=""
              />
            </tr>
          </thead>
          <tbody>
            <tr
              class=""
            >
              <td
                class="table-cell-expand text-left"
              >
                <p
                  class="table-list-title"
                >
                  Item 1
                </p>
              </td>
              <td
                class="table-cell-expand-smaller text-left"
              >
                Object
              </td>
              <td
                class="table-cell-expand-smaller text-left"
              >
                standalone
              </td>
              <td
                class="table-cell-expand-smaller text-left"
              >
                months ago
              </td>
              <td
                class="table-cell-expand-smaller text-left"
              >
                months ago
              </td>
              <td
                class="table-cell-expand-smaller text-left"
              >
                <span
                  class="label label-success"
                >
                  <span
                    class="label-item label-item-expand"
                  >
                    deployed
                  </span>
                </span>
              </td>
              <td
                class=""
              >
                <div
                  class="dropdown dropdown-action"
                >
                  <button
                    class="dropdown-toggle page-link btn btn-unstyled"
                    style=""
                    type="button"
                  >
                    <svg
                      class="lexicon-icon lexicon-icon-ellipsis-v"
                      role="presentation"
                    >
                      <use
                        xlink:href="#ellipsis-v"
                      />
                    </svg>
                  </button>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
